File: frontend/src/types/interview.ts
export type QuestionItem = {
  id: string;
  prompt: string;
  intent?: string;
};

export type InterviewSession = {
  session_id: string;
  interview_title: string;
  questions: Array<{ id: string; prompt: string }>;
  answers: Array<{ question_id: string; response: string }>;
  current_question_index: number;
  is_complete: boolean;
};

export type AnalysisResult = {
  session_id: string;
  summary: string;
  themes: string[];
  recommendations: string[];
};

File: backend/app/schemas/interview.py
from pydantic import BaseModel


class InterviewQuestion(BaseModel):
    id: str
    prompt: str


class AnswerItem(BaseModel):
    question_id: str
    response: str


class StartInterviewRequest(BaseModel):
    interview_title: str
    questions: list[InterviewQuestion]


class InterviewSession(BaseModel):
    session_id: str
    interview_title: str
    questions: list[InterviewQuestion]
    answers: list[AnswerItem]
    current_question_index: int
    is_complete: bool


class AnswerSubmission(BaseModel):
    session_id: str
    question_id: str
    response: str
